src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -192,6 +198,54 @@ final case class AgreementApiServiceImpl(
     }
   }
 
+  override def cloneAgreement(agreementId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route = authorize(ADMIN_ROLE) {
+    val operationLabel = s"Cloning agreement $agreementId"
+    logger.info(operationLabel)
+
+    val result: Future[Agreement] = for {
+      requesterOrgId <- getOrganizationIdFutureUUID(contexts)
+      agreementUUID  <- agreementId.toFutureUUID
+      agreement      <- agreementManagementService.getAgreementById(agreementUUID)
+      _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
+      _              <- Future
+        .failed(AgreementNotInExpectedState(agreement.id.toString, agreement.state))
+        .unlessA(agreement.state == AgreementState.REJECTED)
+      newAgreement   <- agreementManagementService.createAgreement(
+        AgreementSeed(
+          agreement.eserviceId,
+          agreement.descriptorId,
+          agreement.producerId,
+          agreement.consumerId,
+          Nil,
+          Nil,
+          Nil,
+          agreement.consumerNotes
+        )
```

Comment:
Could you please move this creation into [Adapters](https://github.com/pagopa/interop-be-agreement-process/blob/1.0.x/src/main/scala/it/pagopa/interop/agreementprocess/common/Adapters.scala)? It would be great to have it in a method `def asSeed: AgreementSeed` in the implicit class that wraps `Agreement`